### PR TITLE
network-generator: Add logging

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

**Comment by yuwata** [comment]
```
+ SYSTEMD_LOG_LEVEL=debug
+ SYSTEMD_PROC_CMDLINE=ip=foo:dhcp
+ run_network_generator
+ local stderr
+ rm -rf /tmp/test-network-generator-conversion.zHdXH6/run
+ stderr=/tmp/test-network-generator-conversion.zHdXH6/stderr
+ /home/runner/work/systemd/systemd/build/systemd-network-generator --root /tmp/test-network-generator-conversion.zHdXH6
+ [[ -s /tmp/test-network-generator-conversion.zHdXH6/stderr ]]
+ echo 'Generator generated unexpected messages on stderr'
Generator generated unexpected messages on stderr
+ cat /tmp/test-network-generator-conversion.zHdXH6/stderr
Invalid IP address 'foo': Invalid argument
+ return 1
+ rm -rf /tmp/test-network-generator-conversion.zHdXH6
```

**Review by yuwata — APPROVED** [review]
The code change LGTM. Please update the test.

File: src/network/generator/network-generator.c [hunk 1]
```diff
@@ -410,23 +411,32 @@ static int network_set_mtu(Context *context, const char *ifname, const char *mtu
 
         network = network_get(context, ifname);
         if (!network)
-                return -ENODEV;
+                return log_debug_errno(SYNTHETIC_ERRNO(-ENODEV), "No network found for '%s'", ifname);
```
Comment (poettering) [inline src/network/generator/network-generator.c]: drop the `-`

File: src/network/generator/network-generator.c [hunk 2]
```diff
@@ -394,13 +394,14 @@ static int network_set_hostname(Context *context, const char *ifname, const char
...
```
Comment (poettering) [inline src/network/generator/network-generator.c]: -ENODEV → ENODEV

File: src/network/generator/network-generator.c [hunk 3]
```diff
@@ -410,23 +411,32 @@ static int network_set_mtu(Context *context, const char *ifname, const char *mtu
...
+        int r;
 
         assert(context);
         assert(ifname);
         assert(mac);
 
         network = network_get(context, ifname);
         if (!network)
-                return -ENODEV;
+                return log_debug_errno(SYNTHETIC_ERRNO(-ENODEV), "No network found for '%s'", ifname);
```
Comment (poettering) [inline src/network/generator/network-generator.c]: ditto

File: src/network/generator/network-generator.c [hunk 4]
```diff
@@ -443,7 +453,7 @@ static int network_set_address(Context *context, const char *ifname, int family,
...
```
Comment (poettering) [inline src/network/generator/network-generator.c]: ditto

File: src/network/generator/network-generator.c [hunk 5]
```diff
@@ -1220,12 +1234,12 @@ int context_merge_networks(Context *context) {
 
                 r = strv_extend_strv(&network->dns, all->dns, false);
                 if (r < 0)
-                        return r;
+                        return log_oom();
```
Comment (poettering) [inline src/network/generator/network-generator.c]: hmm, log_oom() is on LOG_ERR level. Bu tbelow you log at LOG_DEBUG.

as per coding style we want that functions either log about all errors on their own above LOG_DEBUG, or they do not log about any above LOG_DEBUG (and thus either don#t log at all or stick to LOG_DEBUG) – in the latter case the caller can log instead.

hence decide if this function should log about all errors above LOG_DEBUG or about none, and then stick to it.

log_oom_debug() exists and is your friend btw

**Pull request merged**